docs/timestamps.md

```diff
@@ -0,0 +1,215 @@
+# Timestamps
+
+## Motivation
+In order to enable snapshotting based on time constraints rather than special messages in the Tangle (e.g., milestones), nodes need to share the same perception of time. Specifically, they need to have consensus on the *age of messages*. Therefore, messages contain a field `timestamp` which represents the creation time of the message and is signed by the issuing node. 
```

Comment:
Probably better not to mention milestones (say checkpoints?)
Another application for timestamps is for Adaptive PoW